File: src/main/java/burp/model/GatherMap.java
package burp.model;

import java.util.List;

import burp.reporting.BurpException;
import org.apache.jena.rdf.model.RDFNode;

import burp.model.gathermaputil.SubGraph;

public interface GatherMap {

	boolean isGatherMap();
	
	List<SubGraph> generateGatherMapGraphs(Iteration i, String baseIRI) throws BurpException;

	List<RDFNode> generateTerms(Iteration i, String baseIRI) throws BurpException;
	
}

File: src/main/java/burp/model/GatherMap.kt
package burp.model

import burp.model.gathermap.SubGraph
import org.apache.jena.rdf.model.RDFNode

interface GatherMap {
    fun isGatherMap(): Boolean
    fun generateGatherMapGraphs(i: Iteration, baseIRI: String): List<SubGraph>
    fun generateTerms(i: Iteration, baseIRI: String): List<RDFNode>
}
